src/API/OnboardingTasks.php

```diff
@@ -715,170 +767,252 @@ function( $has_installed, $bundle ) {
-						'time'       => __( '1 minute', 'woocommerce-admin' ),
-					),
-					array(
-						'id'          => 'appearance',
-						'title'       => __( 'Personalize my store', 'woocommerce-admin' ),
-						'content'     => __(
-							'Add your logo, create a homepage, and start designing your store.',
-							'woocommerce-admin'
-						),
-						'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
-						'isComplete'  => get_option( 'woocommerce_task_list_appearance_complete' ),
-						'isVisible'   => true,
-						'time'        => __( '2 minutes', 'woocommerce-admin' ),
-					),
+						)
+					) > 0,
+				'time'        => __( '1 minute', 'woocommerce-admin' ),
+			),
+			array(
+				'id'         => 'marketing',
+				'title'      => __( 'Set up marketing tools', 'woocommerce-admin' ),
+				'content'    => __(
+					'Add recommended marketing tools to reach new customers and grow your business',
+					'woocommerce-admin'
 				),
+				'isComplete' => $has_installed_marketing_extensions,
+				'isVisible'  => Features::is_enabled( 'remote-free-extensions' ) && count( $marketing_extension_bundles ) > 0,
+				'time'       => __( '1 minute', 'woocommerce-admin' ),
+			),
+			array(
+				'id'          => 'appearance',
+				'title'       => __( 'Personalize my store', 'woocommerce-admin' ),
+				'content'     => __(
+					'Add your logo, create a homepage, and start designing your store.',
+					'woocommerce-admin'
+				),
+				'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
+				'isComplete'  => get_option( 'woocommerce_task_list_appearance_complete' ),
+				'isVisible'   => true,
+				'time'        => __( '2 minutes', 'woocommerce-admin' ),
 			),
 		);
 
-		return rest_ensure_response( apply_filters( 'woocommerce_admin_onboarding_tasks', $tasks ) );
+		$snoozed_tasks     = get_option( 'woocommerce_task_list_remind_me_later_tasks' );
+		$tasks_with_snooze = array_map(
+			function ( $task ) use ( $snoozed_tasks ) {
+				$is_snoozed        = in_array( $task['id'], array_keys( $snoozed_tasks ), true );
+				$task['isSnoozed'] = $is_snoozed;
+
+				if ( $is_snoozed ) {
+					$task['snoozedUntil'] = $snoozed_tasks[ $task['id'] ];
+				}
+
+				return $task;
+			},
+			$tasks
+		);
+
+		$tasklist = array(
+			array(
+				'id'         => 'setup',
+				'isComplete' => get_option( 'woocommerce_task_list_complete' ) === 'yes',
+				'isHidden'   => get_option( 'woocommerce_task_list_hidden' ) === 'yes',
+				'title'      => __( 'Get ready to start selling', 'woocommerce-admin' ),
+				'tasks'      => $tasks_with_snooze,
+			),
+		);
+
+		return rest_ensure_response( apply_filters( 'woocommerce_admin_onboarding_tasks', $tasklist ) );
 	}
+
+	/**
+	 * Retrieve single task.
+	 *
+	 * @param String $task_id Task ID.
+	 *
+	 * @return Object
+	 */
+	protected function get_task_by_id( $task_id ) {
+		$tasks_res = $this->get_tasks()->data;
+		$task_list = (object) $tasks_res[0];
```

Comment:
This will only get the first task list, which could create confusion.  Since we can have multiple task lists, I think we'll need to loop over all of them.

Reply:
Good call, fixed 👍🏻